src/main/java/io/github/thebusybiscuit/exoticgarden/PlantsListener.java

```diff
@@ -123,67 +186,102 @@ public void onGenerate(ChunkPopulateEvent e) {
+                                        Rotatable ss = (Rotatable) current.getRelative(BlockFace.UP).getBlockData();
+                                        ss.setRotation(faces[random.nextInt(faces.length)]);
+                                        current.getRelative(BlockFace.UP).setBlockData(ss);
+                                        SkullBlock.setFromHash(current.getRelative(BlockFace.UP), berry.getTexture());
+                                        break;
+                                    default:
+                                        break;
+                                }
+                                break;
+                            }
+                        }
+                    });
                 }
-            }
-            else if (random.nextInt(100) < cfg.getInt("chances.TREE")) {
+            } else if (random.nextInt(100) < cfg.getInt("chances.TREE")) {
                 Tree tree = ExoticGarden.getTrees().get(random.nextInt(ExoticGarden.getTrees().size()));
-
-                plugin.getServer().getScheduler().scheduleSyncDelayedTask(plugin, () -> {
-                    int x = e.getChunk().getX() * 16 + random.nextInt(16);
-                    int z = e.getChunk().getZ() * 16 + random.nextInt(16);
-
+                int x = e.getChunk().getX() * 16 + random.nextInt(16);
+                int z = e.getChunk().getZ() * 16 + random.nextInt(16);
+                if (PaperLib.isChunkGenerated(e.getWorld(), x, z)) {
                     for (int y = e.getWorld().getMaxHeight(); y > 30; y--) {
                         Block current = e.getWorld().getBlockAt(x, y, z);
-
                         if (!current.getType().isSolid() && current.getType() != Material.WATER && current.getType() != Material.SEAGRASS && current.getType() != Material.TALL_SEAGRASS && !(current.getBlockData() instanceof Waterlogged && ((Waterlogged) current.getBlockData()).isWaterlogged()) && tree.isSoil(current.getRelative(0, -1, 0).getType()) && isFlat(current)) {
                             Schematic.pasteSchematic(new Location(e.getWorld(), x, y, z), tree);
                             break;
                         }
                     }
-                });
+                } else {
+                    PaperLib.getChunkAtAsync(e.getWorld(), x, z).thenRun(() -> {
+                        for (int y = e.getWorld().getMaxHeight(); y > 30; y--) {
+                            Block current = e.getWorld().getBlockAt(x, y, z);
+                            if (!current.getType().isSolid() && current.getType() != Material.WATER && current.getType() != Material.SEAGRASS && current.getType() != Material.TALL_SEAGRASS && !(current.getBlockData() instanceof Waterlogged && ((Waterlogged) current.getBlockData()).isWaterlogged()) && tree.isSoil(current.getRelative(0, -1, 0).getType()) && isFlat(current)) {
+                                Schematic.pasteSchematic(new Location(e.getWorld(), x, y, z), tree);
+                                break;
```

Comment:
These two branches do exactly the same, this redundancy is not really a good design, so you should probably split those lines into their own method and just call this method both times.